src/app/components/hris/employees/employee-profile/accordions-documents/accordion-starter-kit-documents/accordion-documents.component.ts

```diff
@@ -0,0 +1,281 @@
+import { Component, EventEmitter, HostListener, Input, Output } from '@angular/core';
+import { EmployeeDocument } from 'src/app/models/hris/employeeDocument.interface';
+import { EmployeeDocumentService } from 'src/app/services/hris/employee/employee-document.service';
+import { Document } from 'src/app/models/hris/constants/documents.contants';
+import { EmployeeProfile } from 'src/app/models/hris/employee-profile.interface';
+import { ActivatedRoute } from '@angular/router';
+import { SnackbarService } from 'src/app/services/shared-services/snackbar-service/snackbar.service';
+import { MatTableDataSource } from '@angular/material/table';
+import { CookieService } from 'ngx-cookie-service';
+import { AuthAccessService } from 'src/app/services/shared-services/auth-access/auth-access.service';
+import { NavService } from 'src/app/services/shared-services/nav-service/nav.service';
+
+@Component({
+  selector: 'app-accordion-documents-starterkit',
+  templateUrl: './accordion-documents.component.html',
+  styleUrls: ['./accordion-documents.component.css']
+})
+export class AccordionDocumentsComponent {
+  @Output() updateDocument = new EventEmitter<number>();
+  @Input() employeeProfile!: EmployeeProfile;
+
+  screenWidth = window.innerWidth;
+
+  @HostListener('window:resize', ['$event'])
+  onResize() {
+    this.screenWidth = window.innerWidth;
+  }
+
+  selectedEmployee!: EmployeeProfile;
+  fileCategories = Document;
+  documentFormProgress: number = 0;
+  documentsProgress: number = 0;
+  employeeDocuments: EmployeeDocument[] = [];
+  documentsFileName: string = "";
+  base64String: string = "";
+  uploadButtonIndex: number = 0;
+  employeeId = this.route.snapshot.params['id'];
+  previousPage: string = '';
+  PREVIOUS_PAGE = "previousPage";
+  showBackButtons: boolean = true;
+  dataSource = new MatTableDataSource<string>();
+  selectedFile !: File;
+  roles: string[] = [];
+  isLoadingUpload: boolean = false;
+
+  constructor(
+    private employeeDocumentService: EmployeeDocumentService,
+    private route: ActivatedRoute,
+    private snackBarService: SnackbarService,
+    private cookieService: CookieService,
+    private authAccessService: AuthAccessService,
+    public navService: NavService,
+  ) { }
+
+  ngOnInit() {
+    this.getEmployeeDocuments();
+    const types: string = this.cookieService.get('userType');
+    this.roles = Object.keys(JSON.parse(types));
```

Comment:
Please try and use the roles from the auth-access service file to detect the role